agent/client.go

```diff
@@ -44,19 +44,28 @@ type client struct {
 
 // NewClient creates a agent RPC client to the given address
 // using the specified client certificate certFile
-func NewClient(addr, certFile string) (*client, error) {
-	cert, err := ioutil.ReadFile(certFile)
+func NewClient(addr, caFile, certFile, keyFile string) (*client, error) {
+	// Load client cert/key
+	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
 	if err != nil {
-		return nil, trace.ConvertSystemError(err)
+		return nil, trace.Wrap(err)
 	}
+
+	// Load the CA of the server
+	clientCACert, err := ioutil.ReadFile(caFile)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
 	certPool := x509.NewCertPool()
-	if !certPool.AppendCertsFromPEM(cert) {
-		return nil, trace.Wrap(err, "failed to append certificates from %v", certFile)
+	if !certPool.AppendCertsFromPEM(clientCACert) {
+		return nil, trace.Wrap(err, "failed to append certificates from %v", caFile)
 	}
 
 	creds := credentials.NewTLS(&tls.Config{
-		RootCAs:    certPool,
-		MinVersion: tls.VersionTLS12,
+		RootCAs:      certPool,
```

Comment:
You need to be able to pass `ServerName` override here as the test certificates use `agent` as a CN.